yazi-dds/src/client.rs

```diff
@@ -62,6 +63,50 @@ impl Client {
 		});
 	}
 
+	/// Connect to an existing server and listen in on the messages that are being
+	/// sent by other yazi instances.
+	/// If no server is running, fail right away.
+	/// If a server is closed, attempt to reconnect forever.
+	pub async fn echo_events_to_stdout(kinds: HashSet<String>) -> Result<()> {
+		let mut lines = Self::connect_listener(&kinds).await?;
+
+		loop {
+			match lines.next_line().await {
+				Ok(Some(s)) => {
+					let kind = s.split(',').next();
+					if matches!(kind, Some(kind) if kinds.contains(kind)) {
+						println!("{}", s);
+					}
+				}
+				Ok(None) => loop {
+					println!("Connection closed");
+					match Self::connect_listener(&kinds).await {
+						Ok(new_lines) => {
+							lines = new_lines;
+							break;
+						}
+						Err(_) => {
+							time::sleep(time::Duration::from_secs(1)).await;
+						}
+					};
+				},
+				Err(e) => {
+					// could not establish initial connection
+					return Err(e.into());
+				}
```

Comment:
```suggestion

```

Why not just `match lines.next_line().await?`?

Reply:
Yeah you're right, this change of mine makes no sense at all 😄 
Will simplify this.

Reply:
Used `match lines.next_line().await.context("Could not establish initial connection")? {` to essentially do the same thing